packages/injected/src/roleUtils.ts

```diff
@@ -178,8 +178,28 @@ const kImplicitRoleByTagName: { [tagName: string]: (e: Element) => AriaRole | nu
       return 'columnheader';
     if (e.getAttribute('scope') === 'row')
       return 'rowheader';
-    const table = closestCrossShadow(e, 'table');
-    const role = table ? getExplicitAriaRole(table) : '';
+    const table = closestCrossShadow(e, 'table') as HTMLTableElement | undefined;
+    if (!table)
+      return 'cell';
+
+    const rows = [...table.rows];
+    const position = getCellPosition(table.rows, e as HTMLTableCellElement);
+
+    if (position) {
+      const { x, y } = position;
+
+      const containingRow = rows[y];
+      if (containingRow) {
+        if (everyNodeIsTH(containingRow.cells))
+          return 'columnheader';
+      }
+
+      const containingColumnCells = rows.map(row => row.cells[x]);
```

Comment:
Could `row.cells[x]` be `null/undefined` when colspan or rowspan is present? Let's be defensive here.